src/scroll-timeline-base.js

```diff
@@ -160,15 +154,23 @@ export function removeAnimation(scrollTimeline, animation) {
   animations.splice(index, 1);
 }
 
+function find(scrollTimeline, animation) {
+  const animations = scrollTimelineOptions.get(scrollTimeline).animations;
+  let index = animations.indexOf(animation);
+  return (index === -1) ? null : animations[index];
+}
+
 /**
- * Attaches a Web Animation instance to ScrollTimeline
+ * Attaches a Web Animation instance to ScrollTimeline.
  * @param scrollTimeline {ScrollTimeline}
  * @param animation {Animation}
  * @param options {Object}
  */
 export function addAnimation(scrollTimeline, animation, options) {
-  let animations = scrollTimelineOptions.get(scrollTimeline).animations;
-  animations.push(animation);
+  if (!find(scrollTimeline, animation)) {
```

Comment:
I'm a bit worried this may cover up errors of cases where we should have removed the animation but failed to - could we throw an error if the animation already exists?